modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/hash/io/HashIndexTreeIo.java

```diff
@@ -67,6 +67,39 @@ public interface HashIndexTreeIo {
             + PARTITIONLESS_LINK_SIZE_BYTES // Index column link.
             + 2 * Long.BYTES; // Row ID.
 
+    /** Index columns are not fully inlined, the index column size is {@link #indexColumnsInlineSize()}. */
+    short NOT_FULLY_INLINE = -1;
+
+    /** Offset of the index columns hash (4 bytes). */
+    int HASH_OFFSET = 0;
+
+    /** Offset of the index columns size (2 bytes). */
+    int SIZE_OFFSET = HASH_OFFSET + Integer.BYTES;
+
+    /** Offset of the index columns tuple (N bytes). */
+    int TUPLE_OFFSET = SIZE_OFFSET + Short.BYTES;
+
+    /**
+     * Returns offset of the index columns link (6 bytes).
+     */
+    default int linkOffset() {
```

Comment:
Should these be private as well?

Reply:
I think it will be a little strange that the offset constants are not private, but the methods are private.
wdyt?

Reply:
Ok, I just see no use for them